components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/FrameworkServiceDataHolder.java

```diff
@@ -63,6 +65,8 @@
     private ClaimMetadataManagementService claimMetadataManagementService = null;
     private SSOConsentService ssoConsentService;
 
+    private JsFunctionRegistryImpl jsFunctionRegistry = new JsFunctionRegistryImpl();
```

Comment:
Would nt it better if we initialize to null and set it at the Component init, The data holder is just a holder, and should not construct any variables.